File: src/cpp/Source/Chunks/Rdef/ShaderInputType.cpp
#include "PCH.h"
#include "ShaderInputType.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(ShaderInputType value)
{
	switch (value)
	{
	case ShaderInputType::CBuffer :
		return "cbuffer";
	case ShaderInputType::Texture:
	case ShaderInputType::Structured:
	case ShaderInputType::ByteAddress:
		return "texture";
	case ShaderInputType::Sampler:
		return "sampler";
	case ShaderInputType::UavRwTyped:
	case ShaderInputType::UavRwStructured:
	case ShaderInputType::UavRwByteAddress:
	case ShaderInputType::UavAppendStructured:
	case ShaderInputType::UavConsumeStructured:
	case ShaderInputType::UavRwStructuredWithCounter:
		return "UAV";
	default:
		throw runtime_error("Shader input type '" + to_string((int) value) + "' is not supported.");
	}
}

File: src/SlimShader/Chunks/Rdef/ShaderInputType.cs
namespace SlimShader.Chunks.Rdef
{
	public enum ShaderInputType
	{
		[Description("cbuffer")]
		CBuffer = 0,

		TBuffer = 1,

		[Description("texture")]
		Texture = 2,

		[Description("sampler")]
		Sampler = 3,

		[Description("UAV")]
		UavRwTyped = 4,

		[Description("texture")]
		Structured = 5,

		[Description("UAV")]
		UavRwStructured = 6,

		[Description("texture")]
		ByteAddress = 7,

		[Description("UAV")]
		UavRwByteAddress = 8,

		[Description("UAV")]
		UavAppendStructured = 9,

		[Description("UAV")]
		UavConsumeStructured = 10,

		[Description("UAV")]
		UavRwStructuredWithCounter = 11
	}
}